### DISCUSSION TITLE
Expanding the 3D skeleton with a fixed radius

### DISCUSSION THREAD

**Discussion by zahrakheyrollahi** [CGAL/cgal#7886] [General]
Hello,I am very happy to have been able to work with a powerful library like cgal.
And the question that came to me is this:
How can I write code to expand the output skeleton in 3D with a fixed radius to get a 3D output similar to the original shape?
I got outputs like this with polyhedron demo so I can save data as .wkt
![elephant11](https://github.com/CGAL/cgal/assets/147084302/ec0c4722-4507-4587-ada5-c419f8f2ab22)

How can I access the edges from these output files and multiply it by a fixed number and make the 3D mesh?

## Source Code

```C++
#include <CGAL/IO/WKT.h>

namespace my_namespace {

class Point3 {
public:
  double x;
  double y;
  double z;

  Point3() {}

  Point3(double x, double y, double z) : x(x), y(y), z(z) {}

  // Add the unit() member function.
  Point3 unit() const {
    return Point3(x / sqrt(x * x + y * y + z * z), y / sqrt(x * x + y * y + z * z), z / sqrt(x * x + y * y + z * z));
  }
};

template <typename T>
class Segment_3 {
public:
  T p1;
  T p2;

  Segment_3() {}

  Segment_3(T p1, T p2) : p1(p1), p2(p2) {}
};

typedef std::vector<Point3> Linestring3;

class MultiLineString3 {
public:
  using Point3 = Point3;
  using Line3 = Segment_3<Point3>;

  std::vector<Linestring3> linestrings;

  // Add the readFromWKT() member function.
  void readFromWKT(const std::string& wkt) {
    // Read the data from the WKT file.
    std::vector<Linestring3> linestrings;
    CGAL::readWKT(wkt, linestrings);
    this->linestrings = linestrings;
  }
};

} // namespace my_namespace

int main() {
  // Declare the mls variable before using it.
  my_namespace::MultiLineString3<double> mls;

  // Get the desired radius from the user.
  double radius;
  std::cout << "Enter the desired radius: ";
  std::cin >> radius;

  // Expand the MultiLineString3 object.
  for (my_namespace::Linestring3& linestring : mls.linestrings) {
    for (my_namespace::Point3& point : linestring) {
      // Use the operator overload for * to multiply the radius and the point.unit() object.
      point = point + radius * point.unit();
    }
  }

  // Read data from WKT file.
  std::string wkt_input;
  std::ifstream wkt_file("input.wkt");
  wkt_file >> wkt_input;

  // Call the readWKT() function.
  mls.readFromWKT(wkt_input);

  // Write the data to the WKT file.
  std::string wkt_output;
  mls.writeToWKT(wkt_output);

  // Save the data to the file.
  std::ofstream wkt_file_out("output.wkt");
  wkt_file_out << wkt_output;

  return 0;
}
```
The errors I get:
![IMG_20231121_220920_451](https://github.com/CGAL/cgal/assets/147084302/d81037fd-d4a7-45ba-8a4c-9342af978d22)
@sloriot 
@lrineau 
## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits):wsl,ubuntu 22.04
* Compiler:g++
* Release or debug mode:gdb
* CGAL version:5.6
* Boost version:1_83_0

**Comment by afabri** [comment]
Please have a look at the[ manual](https://doc.cgal.org/latest/Stream_support/group__PkgStreamSupportIoFuncsWKT.html) for coding.   There is no overload for your call.   Also better put code on gist.github.com so that it is readable.

**Comment by zahrakheyrollahi** [comment]
Hello again, I fixed my code and only this error remains, which I think is due to the CGAL namespace;  How can I solve it?
![IMG_20231122_211413_049](https://github.com/CGAL/cgal/assets/147084302/f69e1ca5-fc6e-4bc3-8d59-a2763248ce24)
If I add cgal.h to my folder, will it be solved?

@afabri

**Comment by afabri** [comment]
Why don't you go to the page I gave you  to have a look which overlaods exist?

**Comment by zahrakheyrollahi** [comment]
Hello again. I was able to fix the errors in my code with the help of your instructions. I only have errors for the headers. How can I solve them?
![h1-1](https://github.com/CGAL/cgal/assets/147084302/d68f11a3-6603-48cf-b8b2-a5b300a50bf3)
@afabri 
@sloriot
